app/Http/Client/Moralis/MoralisPendingRequest.php

```diff
@@ -100,26 +100,29 @@ public function getWalletTokens(Wallet $wallet, Network $network, array $query =
          *  thumbnail: string|null,
          *  decimals: int,
          *  balance: string,
+         *  possible_spam: boolean,
          * }> $data
          **/
         $data = self::get($wallet->address.'/erc20', [
             'chain' => $chain->value,
             ...$query,
         ])->json();
 
-        return collect($data)->map(function ($token) use ($wallet, $network) {
-            return new Web3Erc20TokenData(
-                $token['token_address'],
-                $network->id,
-                $wallet->address,
-                $token['name'],
-                $token['symbol'],
-                $token['decimals'],
-                $token['logo'],
-                $token['thumbnail'],
-                $token['balance'],
-            );
-        });
+        return collect($data)
+            ->filter(fn ($token) => ! $token['possible_spam'])
```

Comment:
Minor improvement could be to use `Arr::get($token, 'possible_spam')` to prevent warnings if the key doesn't exist. This shouldn't happen but who knows if they still randomly serve tokens without this flag.

Reply:
`->reject(fn ($token) => $token['possible_spam'] ?? false)`

:trollface:

Reply:
touché :trollface: